smurfs/src/reducers/index.js

```diff
@@ -13,6 +24,50 @@
    error: null
  }
 */
+export const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case GET_SMURFS_START:
+      return {
+        ...state,
+        fetchingSmurfs: true,
+        error: ""
+      };
+    case GET_SMURFS_SUCCESS:
+      return {
+        ...state,
+        smurfs: action.payload,
+        fetchingSmurfs: false
+      };
+    case GET_SMURFS_FAIL:
+      return {
+        ...state,
+        error: action.payload
+      };
+    case ADD_SMURF_START:
+      return {
+        ...state,
+        addingSmurf: true,
+        error: ""
+      };
+    case ADD_SMURF_SUCCESS:
+      return {
+        ...state,
+        smurfs: action.payload,
```

Comment:
You want to be careful with this. If your add method on the backend only returns a single smurf, you will override all of your smurf data with your payload. A better way to do this would be smurfs: [...state.smurfs, action.payload] this way you preserve your current smurf data and then add the new smurf to the array of smurfs. It will depend on the data you have coming back from the server. So keep that in mind moving forward.